File: app/src/main/java/com/vmenon/mpo/core/persistence/ShowDao.java
package com.vmenon.mpo.core.persistence;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.vmenon.mpo.api.Show;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface ShowDao {

    @Query("SELECT * FROM show where id = :id")
    Show getById(long id);

    @Query("SELECT * FROM show where id = :id")
    LiveData<Show> getLiveById(long id);

    @Insert(onConflict = REPLACE)
    void save(Show show);

    @Query("SELECT * FROM show")
    LiveData<List<Show>> load();

    @Query("SELECT * FROM show WHERE lastUpdate < :comparisonTime")
    List<Show> loadLastUpdatedBefore(long comparisonTime);
}

File: app/src/main/java/com/vmenon/mpo/core/persistence/ShowDao.kt
package com.vmenon.mpo.core.persistence

import androidx.room.Dao
import androidx.room.Query

import com.vmenon.mpo.model.ShowModel

import io.reactivex.Flowable
import io.reactivex.Maybe

@Dao
abstract class ShowDao : BaseDao<ShowModel> {
    @Query("SELECT * FROM show where showName = :name")
    abstract fun getByName(name: String): Maybe<ShowModel>

    @Query("SELECT * FROM show WHERE isSubscribed")
    abstract fun getSubscribed(): Flowable<List<ShowModel>>

    @Query("SELECT * FROM show WHERE isSubscribed AND lastUpdate < :comparisonTime")
    abstract fun getSubscribedAndLastUpdatedBefore(comparisonTime: Long): Maybe<List<ShowModel>>
}
